### DISCUSSION TITLE
Polyhedron_3 is valid but the converted Nef_polyhedron_3 is not valid?

### DISCUSSION THREAD

**Discussion by zfengyan** [CGAL/cgal#6938] [Q&A]
My goal is to perform CSG operations and Minkowski sum on a set of nef polyhedra, but after I successfully built the polyhedra (using polyhedron_builder and delegate() function, I also verified the validation of the polyhedra), the converted nef polyhedra became invalid. Is there any specific reason? My guess is due to some geometry errors since the original building is quite complex.

**Comment by sloriot** [comment]
Do you have a minimal example showing the pb?

**Reply by zfengyan** [reply]
Hi! Thanks for answering! I did a bit research on the web, and have found a possible solution: https://stackoverflow.com/questions/23579190/triangulating-polyhedron-faces-in-cgal

The reason, I guess is -> "because a valid polyhedron cannot be directly
converted into a nef polyhedron because each of the two larger facets
does not have its points exactly on a plane. Thus nef is not able to
consider them as a single face.

A solution is to triangulate the polyhedron before feeding nef with it."

so I tried the approach above:

```cpp
typedef CGAL::Simple_cartesian<float> Kernel;
typedef CGAL::Polyhedron_3<Kernel> Polyhedron_3;

Polyhedron_3 polyhedron = load_my_polyhedron();
CGAL::Polygon_mesh_processing::triangulate_faces(polyhedron);
```

and it works! At lease for my case, after the triangulation, the valid nef polyhedra are obtained.
